src/lib/storage/table.cpp

```diff
@@ -17,53 +17,85 @@
 
 namespace opossum {
 
+Table::Table(const uint32_t chunk_size) : _max_chunk_size(chunk_size) {
+  // On table creation, a first chunk shall be created.
+  _append_new_chunk();
+}
+
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  // TODO - match with piazza result
+  DebugAssert(row_count() == 0, "We currently assume that you can only add columns when no data has been added");
```

Comment:
We should also assert that we only have one chunk. In theory, you could break the current code by creating a new table, adding a new (second) chunk to that and then calling add_column - the first (empty) chunk will not be updated then.

Reply:
Never mind, `_append_new_chunk` is private.